module/VuFind/src/VuFind/ChannelProvider/ListItems.php

```diff
@@ -164,9 +192,13 @@ public function getFromSearch(Results $results, $channelToken = null)
      */
     protected function buildListChannels($channelToken)
     {
+        $lists = $this->getLists(
+            $this->tags,
+            $channelToken ? [$channelToken] : $this->ids,
+            $this->userIds,
+            $this->initialListsToDisplay
+        );
         $channels = [];
-        $lists = $channelToken
```

Comment:
In the original code, if we received a channel token, we would short circuit the whole lookup process and return only that channel. This is needed for AJAX loading specific channels from the drop-down list. It looks like in the new code, the channel token is getting passed to `getLists` along with all the other standard parameters; I think this may result in unexpected/undesired results. Maybe it's okay and I'm just not reading the code closely enough -- but wanted to be sure you understood the context of the channel token, since it may not be obvious at first glance.

Reply:
I tested this quickly in the UI and couldn't find unexpected results, although I might miss something. Overriding $this->ids with channel token (when received) and passing the other filtering parameters should not matter since the list for "Add more channels like this" is also filtered using the same parameters?

Although it should produce the same results, we could refactor `buildListChannel` to ignore filtering parameters when handling channel token:
```
if ($channelToken) {
     $lists = $this->getLists(
         null, [$channelToken], null, $this->initialListsToDisplay
     );
} else {
    $lists = $this->getLists(
        $this->tags,
        $this->ids,
        $this->userIds,
        $this->initialListsToDisplay
    );
}
```

Reply:
My concern is that, if we have a channel token, we know exactly which list ID we need to look up, and we want only that list and not anything else. With the current logic, for example, it looks to me like if we have the displayPublicLists setting turned on, we could potentially retrieve and add a whole bunch of extra public lists that we don't actually need. Additionally, generating the whole complicated SQL query to look up lists just to fetch a known ID is probably going to negatively impact performance. I think it might be worth bringing back the pre-existing `getListsById()` method to cover this special case.

Reply:
I reverted the handling of channelToken and ids to their current version. If tags are used (possibly together with ids), the new `getListsByTagAndId` is used. I agree that it's more clear and safe this way.